### ISSUE TITLE
Can't write index file...

### ISSUE THREAD

**Issue by CobraFlow** [atteo/classindex#25]
I receive an error 

Can't write index file: Attempt to reopen a file for path .....\META-INF\services\com.cf.model.Builder

I am using maven 3.2.1, JRE 1.8.0_25

The java class source files being processed are generated by another processor and implement an interface that I have annotated with @IndexSubclasses. These source files exist in two packages. 

Regards

**Comment by sentinelt** [comment]
Would you be able to write a unit test or provide me with a sample project to reproduce the issue?

**Comment by CobraFlow** [comment]
I have forked the code and have been unable to reproduce.

I generate classes that implement an interface that has the @IndexSubclasses annotation.

I attach the variables tab from Idea showing that the keys for the subclassMap are incorrect. I have modified your code to change the key to all three maps to be String with a value of keyElement.toString(). This works for my situation. Is there a reason to use the TypeElement/PackageElement types as keys?

![image](https://cloud.githubusercontent.com/assets/3617910/5854528/c68c22cc-a226-11e4-8b4a-33fa672d21bf.png)
